### ISSUE TITLE
Unable to add custom models to world file

### ISSUE THREAD

**Issue by sleepydbear** [aws-robotics/aws-robomaker-sample-application-helloworld#58]
I've been trying to add a simple custom model file to the example.world file but i keep getting the following error

`[14:28:37 GMT-0500 (Eastern Standard Time)] [Wrn] [ModelDatabase.cc:212] Unable to connect to model database using [http://models.gazebosim.org//database.config]. Only locally installed models will be available.
[14:28:37 GMT-0500 (Eastern Standard Time)] [Err] [ModelDatabase.cc:414] Unable to download model[model://simulation_app/models/model1]
[14:28:37 GMT-0500 (Eastern Standard Time)] [Err] [SystemPaths.cc:429] File or path does not exist[""]
[14:28:37 GMT-0500 (Eastern Standard Time)] Error [parser.cc:689] Unable to find uri[model://simulation_app/models/model1]` 

I placed the model inside the models folder in the simulation_app 
`simulation_ws/src/simpulation_app/models`

How do I export gazebo_model_path correctly? Thanks in advance

**Comment by mm318** [comment]
Hi @sleepydbear, according to the [Gazebo tutorial](http://gazebosim.org/tutorials?tut=components&cat=get_started#ModelFiles), the environment variable you're looking for is likely to be `GAZEBO_MODEL_PATH`.

I would try (if you're using `bash`)
```
source <gazebo_install_path>/share/gazebo/setup.sh
export GAZEBO_MODEL_PATH=$GAZEBO_MODEL_PATH:<absolute_path_to_app_root>/simulation_ws/src/simpulation_app/models
```

**Comment by mm318** [comment]
Are you having any issues specifically related to this HelloWorld sample application?

**Comment by sleepydbear** [comment]
Thanks for addressing the question. I have previously tried exporting the GAZEBO_MODEL_PATH as you suggested but it does not reflect once I run the simulation job, i.e. the model does not show on gazebo. When i echo GAZEBO_MODEL_PATH in simulation job terminal, it shows 

`/opt/amazon/RoboMakerGazebo/share/gazebo-7/models::/home/robomaker/workspace/applications/simulation-application/dependencies/opt/ros/kinetic/share/turtlebot3_gazebo/models:/home/robomaker/workspace/applications/simulation-application/dependencies/opt/ros/kinetic/share/turtlebot3_gazebo/models/turtlebot3_autorace`

It doesn't reflect the environment variable I appended.

I'm just trying to add a simple .sdf model to the sample application.

While [AWS troubleshooting documentation](https://docs.aws.amazon.com/robomaker/latest/dg/troubleshooting-simulation-jobs.html) mentions this error 
> Are your world file model references correct?

It doesn't say how to solve this problem.

**Comment by mm318** [comment]
Thank you for providing more context.

Perhaps you can take a look at https://github.com/aws-robotics/aws-robomaker-small-house-world and https://github.com/aws-robotics/aws-robomaker-bookstore-world for reference. Take note of the `<export>` tags in their `package.xml` files and `install(DIRECTORY ... models ... DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})` command in their `CMakeLists.txt` files. I take it that you already know how to setup your `.world` file.

This should be the correct way to add models, versus hacking the `GAZEBO_MODEL_PATH` environment variable.

**Comment by sleepydbear** [comment]
Thanks a lot. I got it to work based on your suggestions. I can add any custom models to the robomaker simulation job now. Here's how I got it to work
1. Placed my sdf model in following folder structure `sim_ws/src/sim_app/models`
2. Edited CMakeLists.txt to add models to install `install(DIRECTORY launch models worlds...`
3. Edited package.xml to include   
```
<export>
    <gazebo_ros gazebo_model_path="${prefix}/models"/>
 </export>
```
I didn't have to set GAZEBO_MODEL_PATH apart from what was specified in package.xml
